ocaml-lsp-server/test/e2e/__tests__/ocamllsp-merlinCallCompatible.ts

```diff
@@ -0,0 +1,138 @@
+import outdent from "outdent";
+import * as LanguageServer from "../src/LanguageServer";
+import * as Protocol from "vscode-languageserver-protocol";
+import * as Types from "vscode-languageserver-types";
+
+describe("ocamllsp/merlinCallCompatible", () => {
```

Comment:
I should have mentioned this earlier, but we try not to extend the typescript test suite ever since we've introduced the OCaml LSP client. Do you think you can port these tests to e2e-new?

Reply:
Of course yes!

Reply:
Done in https://github.com/ocaml/ocaml-lsp/pull/1265/commits/2e506c6a8e37b7b0a356c4f07eb469601e371fb0, thanks for the feedback!

Reply:
in https://github.com/xvw/ocaml-lsp/blob/tunneling-merlin-lsp-p2-use-merlin-lib/ocaml-lsp-server/test/e2e-new/merlin_call_compatible.ml#L116, the last test of the test suite, I randomly face to the following error: `"Lev_fiber.Io: already closed"`. Did I miss something?

```
@@ -124,4 +124,31 @@ let%expect_test "errors: warning is disabled" =
   Helpers.test source request;
   [%expect
     {|
-    { "resultAsSexp": false, "result": "{\"class\":\"return\",\"value\":[]}" } |}]
+    (* CR expect_test_collector: This test expectation appears to contain a backtrace.
+       This is strongly discouraged as backtraces are fragile.
+       Please change this test to not include a backtrace. *)
+
+    { "resultAsSexp": false, "result": "{\"class\":\"return\",\"value\":[]}" }
+    detached: /-----------------------------------------------------------------------
+    | Internal error: Uncaught exception.
+    | ("Lev_fiber.Io: already closed",
+    | { source =
+    |     "Raised by primitive operation at Lev_fiber.Io.callstack.(fun) in file \"submodules/lev/lev-fiber/src/lev_fiber.ml\", line 590, characters 31-58\n\
+    |      Called from Lev_fiber.Io.create in file \"submodules/lev/lev-fiber/src/lev_fiber.ml\", line 743, characters 17-29\n\
+    |      Called from Lev_fiber.Io.Lazy_fiber.force.(fun) in file \"submodules/lev/lev-fiber/src/lev_fiber.ml\", line 1047, characters 23-27\n\
+    |      Called from Fiber__Scheduler.exec in file \"fiber/src/scheduler.ml\", line 73, characters 8-11\n\
+    |      Called from Fiber__Scheduler.start in file \"fiber/src/scheduler.ml\", line 216, characters 2-42\n\
+    |      Called from Fiber.run.(fun) in file \"fiber/src/fiber.ml\" (inlined), line 17, characters 28-47\n\
+    |      Called from Lev_fiber.run in file \"submodules/lev/lev-fiber/src/lev_fiber.ml\", line 1264, characters 10-59\n\
+    |      Called from Ocaml_lsp_server.run in file \"ocaml-lsp-server/src/ocaml_lsp_server.ml\", lines 987-989, characters 2-51\n\
+    |      Called from Stdune__Exn_with_backtrace.try_with in file \"otherlibs/stdune/src/exn_with_backtrace.ml\", line 9, characters 8-12\n\
+    |      Called from Dune__exe__Main in file \"ocaml-lsp-server/bin/main.ml\", lines 41-42, characters 6-72\n\
+    |      "
+    | })
+    | Raised at Stdune__Code_error.raise in file "otherlibs/stdune/src/code_error.ml", line 10, characters 30-62
+    | Called from Lev_fiber.Io.with_ in file "submodules/lev/lev-fiber/src/lev_fiber.ml", line 983, characters 10-62
+    | Called from Fiber__Scheduler.exec in file "fiber/src/scheduler.ml", line 73, characters 8-11
+    | Re-raised at Stdune__Exn.raise_with_backtrace in file "otherlibs/stdune/src/exn.ml" (inlined), line 38, characters 27-56
+    | Called from Stdune__Exn_with_backtrace.reraise in file "otherlibs/stdune/src/exn_with_backtrace.ml", line 20, characters 33-71
+    | Called from Fiber__Scheduler.exec in file "fiber/src/scheduler.ml", line 73, characters 8-11
```

Reply:
Usually this happens when you're being sent diagnostics (likely) or code actions (less likely) and you are closing the client before they arrive.

Would be good to see what is being sent in this line:

> Called from Ocaml_lsp_server.run in file \"ocaml-lsp-server/src/ocaml_lsp_server.ml\", lines 987-989, characters 2-51\n\

But regardless, the fix is just to wait for what is being sent to arrive before terminating the test.

I wonder if we should fix ocamllsp to just silently shut itself down on such errors.

Reply:
```ocaml
```

Reply:
@rgrinberg 

> Would be good to see what is being sent in this line:

```ocaml
Lev_fiber.run ~sigpipe:`Ignore (fun () ->
      let* input, output = stream_of_channel channel in
      start (Lsp_fiber.Fiber_io.make input output))
  |> Lev_fiber.Error.ok_exn
```

> But regardless, the fix is just to wait for what is being sent to arrive before terminating the test.

I was inspired by other e2e-test suite, did I miss something about properly wait at the end of the test? Thanks in advance!

Reply:
Currently https://github.com/ocaml/ocaml-lsp/actions/runs/9548374342/job/26316523866?pr=1265 is failing because of that error.